core/src/workflow/machines/workflow_machines.rs

```diff
@@ -477,6 +477,10 @@ impl WorkflowMachines {
                     if let Some(st) = event.event_time {
                         let as_systime: SystemTime = st.try_into()?;
                         self.workflow_start_time = Some(as_systime);
+                        // Set the workflow time to be the event time of the first event, so that
+                        // if there is a query issued before first WFT started event, there is some
+                        // workflow time set.
+                        self.set_current_time(as_systime);
```

Comment:
Didn't go in depth here but we should never set the time unless there's no WFT started event.
Can we ensure that?

Reply:
Why? This will be overridden as soon as there is.

Reply:
Just that it's a potential source for bugs

Reply:
I suppose what I mean to say is this is already doing that. This time can never be used unless there is no WFT started event.

Reply:
Why not only use system time when you know we're in a scenario where there's no WFT start event?